qemu/tests/pci_bridge.py

```diff
@@ -0,0 +1,308 @@
+                                                  pci_bus={'aobject': parent_bus})
+
+    for dev in devices:
+        ret = vm.devices.simple_hotplug(dev, vm.monitor)
+        if ret[1] is False:
+            test.fail("Failed to hotplug device '%s'."
+                      "Output:\n%s" % (dev, ret[0]))
+
+    if drive_format == 'usb2':
+        usb_serial = params["blk_extra_params_%s" % image_name].split("=")[1]
+        res = utils_misc.wait_for(check_usb_in_guest, timeout=360,
+                                  text="Wait for getting usb device info")
+        if res is None:
+            test.fail("Could not find the usb device serial:[%s]" % usb_serial)
+
+    if drive_format == 'virtio':
+        return [devices[-1]]
+    else:
+        return devices[::2]
+
+
+def check_data_disks(test, params, env, vm, session):
+    """
+    Check guest data disks (except image1)
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    :param vm: VM object
+    :param session: VM session
+    """
+    image_list = params.objects("images")
+    del image_list[0]
+    image_num = len(image_list)
+
+    error_context.context("Check data disks in monitor!", logging.info)
+    monitor_info_block = vm.monitor.info_block(False)
+    blocks = ','.join(monitor_info_block.keys())
+    for image in image_list:
+        if image not in blocks:
+            test.fail("drive_%s is missed: %s!" % (image, blocks))
+
+    error_context.context("Read and write on data disks!", logging.info)
+    os_type = params["os_type"]
+    if os_type == "linux":
+        sub_test_type = params.get("sub_test_type", "dd_test")
+        for image in image_list:
+            params["dd_if"] = "ZERO"
+            params["dd_of"] = image
+            utils_test.run_virt_sub_test(test, params, env, sub_test_type)
+
+            params["dd_if"] = image
+            params["dd_of"] = "NULL"
+            utils_test.run_virt_sub_test(test, params, env, sub_test_type)
+    elif os_type == "windows":
+        iozone_cmd = params["iozone_cmd"]
+        iozone_cmd = utils_misc.set_winutils_letter(session, iozone_cmd)
+        data_image_size = params["data_image_size"]
+        disks = utils_disk.get_windows_disks_index(session, data_image_size)
+        disk_num = len(disks)
+        if disk_num < image_num:
+            err_msg = "set disk num: %d" % image_num
+            err_msg += ", get in guest: %d" % disk_num
+            test.fail("Fail to list all the volumes, %s" % err_msg)
+        if not utils_disk.update_windows_disk_attributes(session, disks):
+            test.fail("Failed to update windows disk attributes.")
+        for disk in disks:
+            drive_letter = utils_disk.configure_empty_disk(session, disk,
+                                                           data_image_size,
+                                                           os_type)
```

Comment:
There is an issue in this function, on size handling, so the API is changing here: https://github.com/avocado-framework/avocado-vt/pull/1908, we might remove size parameter here. See also: https://github.com/autotest/tp-qemu/pull/1672. Let's wait for the dependency resolved.

Reply:
Hi, vivian
The size parameter must be passed, you could refer to the fellowing patch:
https://github.com/autotest/tp-qemu/pull/1672
https://github.com/avocado-framework/avocado-vt/pull/1908

If you don't specify the size, you could pass utils_disk.SIZE_AVAILABLE to size.
e.g:
```
            drive_letter = utils_disk.configure_empty_disk(session, disk,
                                                           utils_disk.SIZE_AVAILABLE,
                                                           os_type)
```

Reply:
@YongxueHong Thanks for the information, seems #1672 was just updated. Anyway we better wait for the dependency being merged therefore we have final solution.